### PR TITLE
Avoid some MinGW test failures

### LINKED ISSUE #13558: mingw character set issues?
The log at https://travis-ci.com/github/openssl/openssl/jobs/450013652 contains weird characters and failing tests:
```
C=US, ST=CA, L=San Francisco, O=Example Company, OU=Example Company Unit, CN=Alice Ascii Domain
error 63 at 0 depth lookup: email address mismatch
error ../../test/recipes/25-test_eai_data/ascii_leaf.pem: verification failed
wine ../../apps/openssl.exe verify -nameopt utf8 -no_check_time -verify_email 学生@elementary.school.example.com -CAfile ../../test/recipes/25-test_eai_data/ascii_chain.pem ../../test/recipes/25-test_eai_data/ascii_leaf.pem => 2
not ok 6
```

Note the "学生@elementary.school.example.com"

In the same log file:
```
Could not read certificate from ../../test/shibboleth.pfx
42000000:error:16000071:STORE routines:try_pkcs12:error verifying pkcs12 mac:crypto/store/store_result.c:564:
Unable to load certificate
wine ../../apps/openssl.exe x509 -text -in ../../test/shibboleth.pfx -out out.pem -passin 'pass:σύνθημα γνώρισμα' => 1
not ok 6
```

It's the 64 bit target with those messages, the 32 bit target doesn't have those errors.

### PR DESCRIPTION
**Pull request by mattcaswell**
There were 4 classes of failure:

- line ending problems;
- unicode problems;
- file path munging problems; and
- a "hang" in test_cmp_http.

The unicode problems appear to be somewhere between wine or msys - they
don't actually appear to be a problem with the built binaries. We just skip
those tests for now.

Fixes #13558

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by mattcaswell** [comment]
Temporarily take the approval flags away from this PR. There are still some problems with this for 32-bit MinGW which I am investigating.

**Comment by mattcaswell** [comment]
After much banging-of-head-against-a-brick-wall I finally figured out a workaround for a weird failure in cmp_hdr_test - which only affected 32 bit mingw builds when run on Windows (not wine).

Please take another look at this. This should make both mingw and mingw64 builds work again.

**Review by t8m — APPROVED** [review]

File: test/cmp_hdr_test.c [hunk 1]
```diff
@@ -161,22 +161,32 @@ static int test_HDR_set1_recipient(void)
...
-    struct tm hdrtm;
+    struct tm hdrtm, tmptm;
     time_t hdrtime, before, after, now;
 
     now = time(NULL);
-    before = mktime(gmtime(&now));
+    /*
+     * Trial and error reveals that passing the return value from gmtime
+     * directly to mktime in a mingw 32 bit build gives unexpected results. To
+     * work around this we take a copy of the return value first.
```
Comment (paulidale) [inline test/cmp_hdr_test.c:171]: I wonder if this is related to gmtime() using a static buffer which is overwritten.

**Review by paulidale — APPROVED** [review]

File: test/cmp_hdr_test.c (hunk 1 above)
Comment (mattcaswell) [reply test/cmp_hdr_test.c:171]: It's unclear, but as a single threaded app this shouldn't make a difference.

Comment (paulidale) [reply test/cmp_hdr_test.c:171]: More along the lines of something in mktime overwriting the gmtime buffer.

**Comment by mattcaswell** [comment]
Pushed. Thanks.

File: test/cmp_hdr_test.c (hunk 1 above)
Comment (mattcaswell) [reply test/cmp_hdr_test.c:171]: That's plausible.

Comment (davidben) [reply test/cmp_hdr_test.c:171]: Seems likely. Microsoft's documentation even notes:
> Note that gmtime and localtime (and _gmtime32, _gmtime64, _localtime32, and _localtime64) use a single buffer per thread for the conversion. If you supply this buffer to mktime, _mktime32 or _mktime64, the previous contents are destroyed.

https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mktime-mktime32-mktime64?view=msvc-160

While this doesn't say that you can't pass the buffer to `mktime` (which suggests it's allowed), it does say that the Windows `mktime` touches the buffer. Perhaps the Windows or Visual Studio `mktime` internally makes a copy to account for this, but the Wine or MinGW doesn't?

**Pull request closed without merging**